app/src/app/components/contents/PacketGroup/PacketDataTable.tsx

```diff
@@ -0,0 +1,35 @@
+import { useState } from "react";
+import { PAGE_SIZE } from "../../../../lib/constants";
+import { Packet } from "../../../../types";
+import { DataTable } from "../common/DataTable";
+import { setupPacketColumns } from "./packetColumns";
+import { ToggleParamColumnsDropdownColumns } from "./ToggleParamColumnsDropdown";
+import { initializeColumnVisibility } from "./utils/initializeColumnVisibility";
+
+interface PacketDataTableProps {
+  packets: Packet[];
+}
+export const PacketDataTable = ({ packets }: PacketDataTableProps) => {
+  const allParametersKeys = new Set(packets.flatMap((packet) => Object.keys(packet.parameters)));
+  const [columnVisibility, setColumnVisibility] = useState<Record<string, boolean>>(
+    initializeColumnVisibility(allParametersKeys)
+  );
+  return (
+    <div className="space-y-4 flex flex-col">
+      <div className="flex justify-end">
+        <ToggleParamColumnsDropdownColumns
+          allParametersKeys={allParametersKeys}
+          columnVisibility={columnVisibility}
+          setColumnVisibility={setColumnVisibility}
+        />
+      </div>
+      <DataTable
+        columns={setupPacketColumns(allParametersKeys)}
+        data={packets}
+        pagination={{ pageSize: PAGE_SIZE }}
+        visibility={{ columnVisibility, setColumnVisibility }}
```

Comment:
Ah, so all possible columns are coming from `packetColumns`, but we only show the required parameter columns, with the `visibility` prop here..?

Reply:
sorry my comment didnt save.. but all possible columns actually is the keys of parameters.. and we only show the ones that are toggled via visibility prop yes